### PR TITLE
Support date range in block window

### PR DESCRIPTION
**Pull request by mpolden**
@bratseth

### CODE REVIEW

File: config-model-api/src/test/java/com/yahoo/config/application/api/TimeWindowTest.java [hunk 1]
```diff
@@ -25,71 +28,106 @@
...
+
+            TimeWindow tw2 = TimeWindow.from("sun", "1", "CET", "2022-01-01", "2022-01-02");
+            Instant i6 = Instant.parse("2022-01-01T00:00:00.00Z"); // Wrong day
+            Instant i7 = Instant.parse("2022-01-02T01:00:00.00Z"); // Wrong hour because of time zone offset
+            Instant i8 = Instant.parse("2022-01-02T00:00:00.00Z");
+            assertOutside(tw2, i6);
+            assertOutside(tw2, i7);
+            assertInside(tw2, i8);
+
+            TimeWindow tw3 = TimeWindow.from("", "", "CET", "2022-01-02", "");
```
Comment (freva) [inline config-model-api/src/test/java/com/yahoo/config/application/api/TimeWindowTest.java:83]: Its not possible to define any range shorter than 7 days without also specifying `days`

**Review by bratseth — APPROVED** [review]
Great!

**Review by freva — COMMENTED** [review]
We expose the hours and days block window in application API which is used in console, f.ex.:
image

https://github.com/vespa-engine/vespa/blob/a3bc6f9b6af95b97a2b55147a586ca19d1d98938/controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java#L1162
I think we should change this to be based on _this_ week (as opposed to what it is now, which is the days/hours blocked without date range).

File: config-model-api/src/test/java/com/yahoo/config/application/api/TimeWindowTest.java (hunk 1 above)
Comment (mpolden) [reply config-model-api/src/test/java/com/yahoo/config/application/api/TimeWindowTest.java:83]: Thanks! Fixed in https://github.com/vespa-engine/vespa/pull/20699

**Pull request merged**